specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -214,27 +221,83 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
     </ButtonComponent>
   );
 
+  const [carryForwardAmount, setCarryForwardAmount] = React.useState<number>(1);
+
   return (
     <>
       {typeof handleAdd === 'function' && canCreate ? (
         <>
-          {showClone &&
-            copyButton(
-              formsText.clone(),
-              formsText.cloneDescription(),
-              async () => resource.clone(true)
-            )}
+          {resource.specifyTable.name === 'CollectionObject' &&
+          (isInRecordSet === false || isInRecordSet === undefined) &&
+          isSaveDisabled &&
+          showCarry &&
+          showBulkCarry ? (
+            <Input.Integer
+              aria-label={formsText.bulkCarryForwardCount()}
+              className="!w-fit"
+              min={1}
+              placeholder="1"
+              value={carryForwardAmount}
+              onValueChange={(value): void =>
+                Number.isNaN(value)
+                  ? setCarryForwardAmount(1)
+                  : setCarryForwardAmount(Number(value))
+              }
+            />
+          ) : null}
           {showCarry &&
             copyButton(
               formsText.carryForward(),
               formsText.carryForwardDescription(),
-              async () => resource.clone(false)
+              resource.specifyTable.name === 'CollectionObject' &&
+                carryForwardAmount > 2
+                ? async () => {
+                    const formatter =
+                      tables.CollectionObject.strictGetLiteralField(
+                        'catalogNumber'
+                      ).getUiFormatter()!;
+                    const wildCard = formatter.valueOrWild();
+
+                    const clonePromises = Array.from(
+                      { length: carryForwardAmount },
+                      async () => {
+                        const clonedResource = await resource.clone(false);
+                        await clonedResource.set(
+                          'catalogNumber',
+                          wildCard as never
+                        );
```

Comment:
This makes it so that trying to bulk carry forward catalog numbers with a non-incrementing format will result in an error.

Consider the two following formats for catalognumber:

```xml
  <format system="false" name="CatalogNumberTest" class="edu.ku.brc.specify.datamodel.CollectionObject" fieldname="catalogNumber">
    <field type="year" size="4" value="YEAR" byyear="true"/>
    <field type="separator" size="1" value="-"/>
    <field type="numeric" size="3" inc="false"/>
  </format>
```

In this case, the `wildCard` will be `2024-###`, which is an invalid catalognumber and the backend throws an exception stating this: 

`FormatMismatch at /api/specify/bulk/collectionobject/ value '2024-###' doesn't match formatter YEAR-### `

```xml
  <format system="false" name="CatalogNumberRegex" class="edu.ku.brc.specify.datamodel.CollectionObject" fieldname="catalogNumber" default="true">
    <field type="regex" size="6" minsize="4" value="[0-9]{4}[a-zA-Z]{0,2}" pattern="####[AA]"/>
  </format>
```
In this case, `wildCard` will be the `value` of the regex itself and the backend will fail with a similar exception: 

`FormatMismatch at /api/specify/bulk/collectionobject/ value '[0-9]{4}[a-zA-Z]{0,2}' doesn't match formatter [0-9]{4}[a-zA-Z]{0,2}`

Reply:
I am not sure how we'd want to resolve this. 
(Besides disabling this feature for those cases, but that is far from the ideal). 

I would imagine an ideal workflow might be to navigate to a RecordSet-like interface with the cloned records (without saving them to the backend) when they wish to bulk carry forward. 
And having a button (which could be the normal Save Button), which only allows saving when all records have no save blockers. 

Do we know if Specify 6 handles these cases?

Reply:
> In this case, the wildCard will be 2024-###, which is an invalid catalognumber and the backend throws an exception stating this:

Isn't that a valid catalog number though? Pretty sure that's accepted....

I think it is rare enough for users to want to use this in the first place..... makes it even more unlikely for rege ones..

Reply:
> Isn't that a valid catalog number though? Pretty sure that's accepted....

No, it's not accepted by the backend or frontend. 
In the format, that part of the formatter is explicitly defined as a non-incrementing numeric field. 
The pattern for the field remains `2024-###`, but is marked as invalid and only numeric digits are allowed. 

If it's supposed to be accepted, or was at some point, then should the backend just treat `###` as it would any auto-incrementing numeric formatter? 

--- 

> I think it is rare enough for users to want to use this in the first place..... makes it even more unlikely for rege ones..

Probably a fair point. I am unsure about how many people used this feature in Specify 6 or requested this (apart from the mentions in #545 and #3238), though I do know that Entomologists would most likely use and benefit from this feature. 

At the very least, instead of erroring we should disable this feature then. 

I'll take a look into how Specify 6 handles these cases.

Reply:
Ah, I see that is not incremented now. I don't think it is worth considering. Worse edge cases have not been handled elsewhere